src/v/cluster/archival_metadata_stm.h

```diff
@@ -293,10 +294,13 @@ class archival_metadata_stm final : public persisted_stm<> {
 
     // The offset of the last mark_clean_cmd applied: if the manifest is
     // clean, this will equal _insync_offset.
-    model::offset _last_clean_at;
+    model::offset _last_clean_at{};
 
     // The offset of the last record that modified this stm
-    model::offset _last_dirty_at;
+    model::offset _last_dirty_at{};
+
+    // Offset of the last applied replace_manifest command.
+    std::optional<model::offset> _last_applied_replace_cmd{std::nullopt};
```

Comment:
this is not persisted to the snapshot and will be reset after restart

Reply:
I think this is okay though, since if we have a snapshot, we wouldn't need to replay it

Reply:
In any case, moved to using _insync_offset

Reply:
Actually I don't think we can use _insync_offset since it gets bumped on every batch type, including config batches